app/controllers/webhooks/whatsapp_controller.rb

```diff
@@ -2,6 +2,12 @@ class Webhooks::WhatsappController < ActionController::API
   include MetaTokenVerifyConcern
 
   def process_payload
+    if inactive_whatsapp_number?
+      Rails.logger.warn("Rejected webhook for inactive WhatsApp number: #{params[:phone_number]}")
+      render json: { error: 'Inactive WhatsApp number' }, status: :service_unavailable
```

Comment:
I would suggest to go with unprocessable_entity rather than service unavailable.
```suggestion
      render json: { error: 'Inactive WhatsApp number' }, status: :unprocessable_entity
```